cmd/snap/cmd_interface.go

```diff
@@ -0,0 +1,166 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+
+	"github.com/jessevdk/go-flags"
+)
+
+type cmdInterface struct {
+	ShowAttrs   bool `long:"attrs"`
+	ShowAll     bool `long:"all"`
+	Positionals struct {
+		Interface interfaceName `skip-help:"true"`
+	} `positional-args:"true"`
+}
+
+var shortInterfaceHelp = i18n.G("Lists interfaces in the system")
+var longInterfaceHelp = i18n.G(`
+The interface command lists interfaces available in the system.
+
+By default a list of all used interfaces, along with a short summary, is
+displayed. Use the --all option to include unused interfaces.
+
+$ snap interfaces [--attrs] <interface>
```

Comment:
This and the paragraph below may be dropped. It's just restating the documentation of --attrs.